sw/airborne/modules/nav/nav_rotorcraft_hybrid.c

```diff
@@ -118,7 +123,13 @@ static void nav_hybrid_goto(struct EnuCoor_f *wp)
 
   VECT2_COPY(nav.speed, speed_sp);
   nav.horizontal_mode = NAV_HORIZONTAL_MODE_WAYPOINT;
-  nav.setpoint_mode = NAV_SETPOINT_MODE_SPEED;
+
+  // In optitrack tests use position mode for more precise hovering
+  if (NAV_OPTITRACK){
+    nav.setpoint_mode = NAV_SETPOINT_MODE_POS;
+  }else{
+    nav.setpoint_mode = NAV_SETPOINT_MODE_SPEED;
+  }
```

Comment:
I really don't like to have explicit coupling between control and estimation.
- If you are only using the `goto` function from `nav_hybrid`, then use directly use the default rotorcraft nav and not hybrid nav when flying indoor with optitrack
- If you are really using the hybrid specific function for `route` and `circle`, but not for `goto`, make a configuration parameter (like `NAV_HYBRID_GOTO_MODE`) that defaults to `NAV_SETPOINT_MODE_SPEED`, but please no reference to optitrack or any other positioning strategy here.

Reply:
Thank you for the suggestion, I have implemented the second option.